akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/BehaviorSetup.scala

```diff
@@ -121,3 +121,43 @@ private[akka] object MDC {
     Map("persistenceId" -> persistenceId.id, "phase" -> phaseName)
   }
 }
+
+/**
+  * Setup snapshot and event delete/retention behavior.
+  *
+  * INTERNAL API
+  */
+@InternalApi
+private[persistence]
+final case class RetentionCriteria private(snapshotEveryNEvents: Long,
+                                           keepNSnapshots: Long,
+                                           deleteMessagesOnSnapshot: Boolean) {
+  /**
+    * Delete Messages:
+    *   {{{ toSequenceNr - keepNSnapshots * snapshotSize }}}
+    * Delete Snapshots:
+    *   {{{ (toSequenceNr - 1) - (keepNSnapshots * snapshotSize) }}}
+    *
+    * @param toSequenceNr the sequence number to delete to if `retainSnapshotsOnly` is false
+    */
+  def toSequenceNumber(toSequenceNr: Long): Long = {
+    // Delete old events, retain the latest
+    toSequenceNr - (keepNSnapshots * snapshotEveryNEvents)
+  }
+}
+
+/**
+  * INTERNAL API
+  */
+@InternalApi
+private[persistence] object RetentionCriteria {
+
+  def apply(snapshotEveryNEvents: Long, keepNSnapshots: Long, deleteMessagesOnSnapshot: Option[Boolean]): RetentionCriteria =
+    RetentionCriteria(snapshotEveryNEvents, keepNSnapshots, deleteMessagesOnSnapshot.getOrElse(false))
+
+  def apply(): RetentionCriteria =
+    apply(
+      snapshotEveryNEvents = 1000L,
+      keepNSnapshots = 2L,
+      deleteMessagesOnSnapshot = Some(false))
+}
```

Comment:
Since it is immutable, we can keep the default as a value and reuse that instead of creating a new one for every event sourced behavior.